### PR TITLE
Get rid of shared state for LocalMaxPeakFinder

### PR DESCRIPTION
**Pull request by ttung**
Shared state is fundamentally incompatible with multithreading.

1. Each tile computes its own threshold.
2. Don't share the spot coords.

Part of #834

### CODE REVIEW

File: starfish/core/spots/DetectSpots/local_max_peak_finder.py [hunk 1]
```diff
@@ -205,8 +193,10 @@ def _compute_threshold(self, img: Union[np.ndarray, xr.DataArray]) -> float:
         """
         img = np.asarray(img)
         thresholds, spot_counts = self._compute_num_spots_per_threshold(img)
-        threshold = self._select_optimal_threshold(thresholds, spot_counts)
-        return threshold
+        if len(spot_counts) == 0:
```
Comment (dganguli) [inline starfish/core/spots/DetectSpots/local_max_peak_finder.py]: nice catch!

File: starfish/core/spots/DetectSpots/local_max_peak_finder.py [hunk 2]
```diff
@@ -70,13 +70,6 @@ def __init__(
         self.is_volume = is_volume
         self.verbose = verbose
 
-        # these parameters are useful for debugging spot-calls
```
Comment (dganguli) [inline starfish/core/spots/DetectSpots/local_max_peak_finder.py]: so. these parameters are indeed useful for debugging spot-calls. is there any way to save them, like for example if you're not using multiprocessing? or is state just impossible to have?

Comment (dganguli) [reply starfish/core/spots/DetectSpots/local_max_peak_finder.py]: debugging is the wrong word. these parameters, per tile, are critical to doing a quality control analysis of the spot calls.

Comment (dganguli) [reply starfish/core/spots/DetectSpots/local_max_peak_finder.py]: but on the other hand, this piece of the code should probably be pulled out of this spot detector and made general for use in all of the other spot detectors. @ambrosejcarr @shanaxel42 and i have talked about this. as such, i suggest we punt this issue over to the spot-finding epic.

**Comment by ttung** [comment]
> Does this cause a significant slowdown?

I haven't tested that.

**Pull request merged**